### PR TITLE
[MM-42715] Add icons and shortcuts to mobile webview

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary
This PR adds icons and shortcuts for two mobile webview actions in the `...` menu.

How to test Mobile webiew: 
* reduce window in the x direction until mobile webview is triggered. I have setup a test server in this PR for testing.

Differences include:
* `Add Reaction` - only add an icon
* `Save` option visible
	* clicking will save the post
	* `S` shortcut will save the post
	* icon is outlined
* `Remove from Saved`
	* clicking will unsave the post
	* `S` shortcut will unsave the post
	* icon is filled

<table>
    <tr>
	    <td>Before 
  	    <td>After: post is currently saved
		<td>After: post is currently not saved
    <tr>
  	    <td>image
  	    <td>image
  	    <td>image

</table>

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-42715

#### Release Note

```release-note

```

### CODE REVIEW

**Review by jwilander — APPROVED** [review]

**Review by matthewbirtch — APPROVED** [review]

**Review by Willyfrog — APPROVED** [review]
LGTM!

**Comment by amyblais** [comment]
@DHaussermann This is a bug fix for Cloud March 30th release.

**Comment by jfrerich** [comment]
@DHaussermann its a relatively simple change.

Not related to this PR... @amyblais the remaining tickets for the release fall into two groups that will get fixed by two PRs. When is the last and time to get those in for the release?

**Comment by amyblais** [comment]
>

**Comment by amyblais** [comment]
> @DHaussermann its a relatively simple change.
> 
> Not related to this PR... @amyblais the remaining tickets for the release fall into two groups that will get fixed by two PRs. When is the last and time to get those in for the release?

@jfrerich The deadline for bug fixes for the Cloud release is end of Monday. If a few bug fixes don't make it to the Cloud March 30th release, we can cherry-pick them separately to the v6.6 branch once it has been cut. But ideally we'd like to fix as many bugs as we can for the Cloud release.

(Sorry, I didn't mean to close the PR)

**Comment by jfrerich** [comment]
I thought release was Wednesday and that would be the last day to cherry-pick to v6.6.

No worries either way.  I'll get them in

**Comment by amyblais** [comment]
> I thought release was Wednesday and that would be the last day to cherry-pick to v6.6.
> 
> No worries either way. I'll get them in

Thank you Jason!

**Comment by DHaussermann** [comment]
@jfrerich On the test server in mobile web view - All the post pop-overs show at once when none should have focus. 
Does not happen in the regular webapp view. 

image

Please let me know if you can repro this. It will look this way now if you login. Pretty sure all I did was click the Actions option once to get it in this state. (Reach out if you need help getting it in this state from Fresh)

**Comment by matthewbirtch** [comment]
> @DHaussermann, I don't think we can solve this on mobile view. The issue is that all posts always show the `...` and `actions` menus because there is no hover, like the web view.
> 
> @matthewbirtch, this is a pretty ugly UX issue, but I don't have a workaround for this side effect. Do you have any thoughts?

@jfrerich I saw we just kill the pulsing dot for mobile webview

**Comment by jfrerich** [comment]
@DHaussermann, I'll fix the mobile view tooltip issue in another PR. Please omit that issue for this PR testing

**Comment by DHaussermann** [comment]
@jfrerich sorry I mis understood and was waiting for a commit here.  

Seems this should be closed  on favor of https://github.com/mattermost/mattermost-webapp/pull/10080

cc @saturninoabril

**Comment by saturninoabril** [comment]
@DHaussermann As mentioned offline, I think the linked PR is different as it fixes pulsating dot while this is about showing icons and shortcuts in mobile view. Let me know if that's correct.

File: components/dot_menu/__snapshots__/dot_menu.test.tsx.snap [hunk 1]
```diff
@@ -74,6 +74,11 @@ exports[`components/dot_menu/DotMenu should match snapshot, canDelete 1`] = `
...
     >
       <MenuItemAction
+        icon={
+          <span
+            className="icon-emoticon-plus-outline MenuItem__compass-icon"
+          />
+        }
         onClick={[Function]}
         show={false}
         text="Add Reaction"
```
Comment (saturninoabril) [inline components/dot_menu/__snapshots__/dot_menu.test.tsx.snap:84]: Is there a shortcut key for "Add Reaction"?

File: components/dot_menu/__snapshots__/dot_menu.test.tsx.snap [hunk 2]
```diff
@@ -112,12 +117,24 @@ exports[`components/dot_menu/DotMenu should match snapshot, canDelete 1`] = `
       text="Mark as Unread"
     />
     <MenuItemAction
+      icon={
+        <span
+          className="icon-bookmark MenuItem__compass-icon"
+        />
+      }
       onClick={[Function]}
+      rightDecorator="S"
```
Comment (saturninoabril) [inline components/dot_menu/__snapshots__/dot_menu.test.tsx.snap:126]: Likely,
```suggestion
      rightDecorator={<ShortcutKey shortcutKey='S'/>}
```
same with others.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
@jfrerich Thanks!
Below is what I've got with "Add Reaction" and "Save".
Screen Shot 2022-03-29 at 8 26 10 AM

Other issue: If I opened a menu either with `...` or `Actions` and then close by clicking/pressing anywhere, related thread opened unexpectedly .

File: components/dot_menu/__snapshots__/dot_menu.test.tsx.snap (hunk 1 above)
Comment (jfrerich) [reply components/dot_menu/__snapshots__/dot_menu.test.tsx.snap:84]: Matt suggest to have no shortcut for `Add Reaction`

File: components/dot_menu/__snapshots__/dot_menu.test.tsx.snap (hunk 2 above)
Comment (jfrerich) [reply components/dot_menu/__snapshots__/dot_menu.test.tsx.snap]: This suggestion is correct.  Fixed!

**Comment by saturninoabril** [comment]
> Other issue: If I opened a menu either with ... or Actions and then close by clicking/pressing anywhere, related thread opened unexpectedly.

fyi, this is confirmed as not related to More Actions feature and it's been like that for so long. Ticket filed to address separately - https://mattermost.atlassian.net/browse/MM-42863

File: components/dot_menu/dot_menu.tsx [hunk 3]
```diff
@@ -371,6 +373,12 @@ export class DotMenuClass extends React.PureComponent<Props, State> {
             this.props.handleDropdownOpened(false);
             break;
 
+        // save / unsave
+        case Utils.isKeyPressed(e, Constants.KeyCodes.S):
+            this.handleFlagMenuItemActivated(e);
+            this.props.handleDropdownOpened(false);
+            break;
```
Comment (saturninoabril) [inline components/dot_menu/dot_menu.tsx:380]: @jfrerich I'm seeing an inconsistently working shortcut keys in post menu, especially if you triggered particular key more than once. Like in a case where you saved a post but then change to unsaved.
Eliminating code below at `onShortcutKeyDown` seemed to solve the issue but I wonder why it's placed in the first place.
```javascript
if (this.keysHeldDown.includes(e.key)) {
    return;
}
this.keysHeldDown.push(e.key);
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
@jfrerich All good now except for last comment on inconsistency with shortcut key. Can be addressed separately if you wish but I suggest to include it here.

**Comment by jfrerich** [comment]
@saturninoabril the shortcut key was fixed in the last commit. Do you not see it fixed? ☹️

**Comment by saturninoabril** [comment]
It worked but not consistently if you do it more than once like in the case of save then unsave or pin then unpin.

**Comment by jfrerich** [comment]
And the issue is the shortcut key is shifted left and out of alignment?

**Comment by saturninoabril** [comment]
The UI rendering is perfect but I meant the way shortcut key works is not consistently doing as expected.
Repro:
1. Open post menu by clicking `...` then press "S"
- observed that post is saved
2. Without moving the mouse, click `...` then press "S" again to unsave
- observed that post is not unsaved. It remained saved. Note that sometimes it worked and sometimes not. So, you may repro by trying multiple times.

**Comment by jfrerich** [comment]
@saturninoabril, I can confirm the issue. It is happens on shortcuts that share the same key.  (ie. `S' or `P`). If the menu is open and you use the `S` or `P` shortcut twice it will work. I won't be able to get that small fix in tonight and will start again early tomorrow.

**Review by saturninoabril — APPROVED** [review]
Thanks @jfrerich! Approving now.

Filed https://mattermost.atlassian.net/browse/MM-42892 to address separately for v6.6.

**Pull request merged**